community/consistency-check/src/test/java/org/neo4j/consistency/ConsistencyCheckToolTest.java

```diff
@@ -205,10 +209,35 @@ public void shouldExitWhenRecoveryNeededButRecoveryFalseOptionSpecified() throws
         monitors.addMonitorListener( listener );
 
         ConsistencyCheckTool.ExitHandle exitHandle = mock( ConsistencyCheckTool.ExitHandle.class );
-        StoreRecoverer storeRecoverer = mock( StoreRecoverer.class );
-        when( storeRecoverer.recoveryNeededAt( any( File.class ) ) ).thenReturn( true );
+        DefaultFileSystemAbstraction fs = new DefaultFileSystemAbstraction();
+        File dataDir = storeDirectory.graphDbDir();
+
+        PhysicalLogFiles logFiles = new PhysicalLogFiles( dataDir, fs );
+        File log = logFiles.getLogFileForVersion( logFiles.getHighestLogVersion() );
+        try ( StoreChannel channel = fs.create( log ) )
+        {
+            byte[] message = new String( "A lot of stuff that certanly isn't just the trailer, to induce " +
+                                         "the need for recovery. " ).getBytes();
+            ByteBuffer buffer = ByteBuffer.allocate( message.length );
+            buffer.put( message );
+            buffer.flip();
```

Comment:
You could also make the ByteBuffer wrap the message array, so you don't need to put and flip.